python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,612 @@
+                        state_fip, county_fip)], age, sex, whi, wohi, total])
+
+                # County-Race
+                else:
+                    county_race_data.append([state_fip, self.state_fips[state_fip], county_fip, self.county_fips[(
+                        state_fip, county_fip)], age, race, whi, wohi, total])
+
+        # Build Panda DataFrames with standardized cols
+        self.state_sex_frame = pd.DataFrame(state_sex_data, columns=[
+            STATE_FIPS_COL,
+            STATE_NAME_COL,
+            AGE_COL,
+            SEX_COL,
+            WITH_HEALTH_INSURANCE_COL,
+            WITHOUT_HEALTH_INSURANCE_COL,
+            TOTAL_HEALTH_INSURANCE_COL])
+        self.state_race_frame = pd.DataFrame(state_race_data, columns=[
+            STATE_FIPS_COL,
+            STATE_NAME_COL,
+            AGE_COL,
+            RACE_COL,
+            WITH_HEALTH_INSURANCE_COL,
+            WITHOUT_HEALTH_INSURANCE_COL,
+            TOTAL_HEALTH_INSURANCE_COL])
+        self.county_sex_frame = pd.DataFrame(county_sex_data, columns=[
+            STATE_FIPS_COL,
+            STATE_NAME_COL,
+            COUNTY_FIPS_COL,
+            COUNTY_NAME_COL,
+            AGE_COL,
+            SEX_COL,
+            WITH_HEALTH_INSURANCE_COL,
+            WITHOUT_HEALTH_INSURANCE_COL,
+            TOTAL_HEALTH_INSURANCE_COL])
+        self.county_race_frame = pd.DataFrame(county_race_data, columns=[
+            STATE_FIPS_COL,
+            STATE_NAME_COL,
+            COUNTY_FIPS_COL,
+            COUNTY_NAME_COL,
+            AGE_COL,
+            RACE_COL,
+            WITH_HEALTH_INSURANCE_COL,
+            WITHOUT_HEALTH_INSURANCE_COL,
+            TOTAL_HEALTH_INSURANCE_COL])
+
+        # Aggregate Frames by Filename
+        self.frames = {
+            'table_health_insurance_by_race_state': self.state_race_frame,
+            'table_health_insurance_by_sex_state': self.state_sex_frame,
+            'table_health_insurance_by_race_county': self.county_race_frame,
+            'table_health_insurance_by_sex_county': self.county_sex_frame
+        }
+
+
+class ACSHealthInsurance(DataSource):
+
+    @staticmethod
+    def get_id():
+        """Returns the data source's unique id. """
+        return 'ACS_HEALTH_INSURANCE'
+
+    # Uploads to GCS. Sees if the data has changed by diffing the old run vs the new run.
+    # (presumably to skip the write to bq step though not 100% sure as of writing this)
+    def upload_to_gcs(self, gcs_bucket, **attrs):
+        file_diff = False
+        for ingester in self._create_ingesters():
+            next_file_diff = ingester.upload_to_gcs(gcs_bucket)
+            file_diff = file_diff or next_file_diff
+        return file_diff
+
+    def write_to_bq(self, dataset, gcs_bucket, **attrs):
+        for ingester in self._create_ingesters():
+            ingester.write_to_bq(dataset, gcs_bucket)
+
+    def _create_ingesters(self):
```

Comment:
Is this method necessary as it only returns a single element list?

Reply:
Im just following the flow from ACS_population, and could be split out in the future